pkg/http/sender.go

```diff
@@ -0,0 +1,605 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package http
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/questdb/go-questdb-client/v3/pkg/buffer"
+	"github.com/questdb/go-questdb-client/v3/pkg/conf"
+)
+
+type tlsMode int64
+
+const (
+	tlsDisabled           tlsMode = 0
+	tlsEnabled            tlsMode = 1
+	tlsInsecureSkipVerify tlsMode = 2
+)
+
+var (
+	// We use a shared http transport to pool connections
+	// across LineSenders
+	globalTransport http.Transport = http.Transport{
+		Proxy:                 http.ProxyFromEnvironment,
+		MaxConnsPerHost:       0,
+		MaxIdleConns:          100,
```

Comment:
We should expose this as a sender setting. To achieve that, we could create the transport lazily, i.e. on the first sender initialization.

Reply:
What about making a `WithTransport` option for users to customize their sender's transport? We would default to the global transport.  

This would also be a nice workaround for the global WithTlsInsecureSkipVerify setting problem (once it's set, it can't be unset because we only have a single tls config for the globalTransport). If you want multiple senders with different tls configs, you can do that with a custom transport option.  Wdyt?